extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueLightHandler.java

```diff
@@ -122,13 +124,19 @@ public void handleCommand(ChannelUID channelUID, Command command) {
         }
 
         StateUpdate lightState = null;
-
         switch (channelUID.getId()) {
             case CHANNEL_COLORTEMPERATURE:
                 if (command instanceof PercentType) {
                     lightState = LightStateConverter.toColorTemperatureLightState((PercentType) command);
                 } else if (command instanceof OnOffType) {
                     lightState = LightStateConverter.toOnOffLightState((OnOffType) command);
+                    /*
+                     * Call OSRAM PAR16 50 workaround code if bulb is of that type
+                     */
+                    if (HueBindingConstants.THING_TYPE_PAR16_50_TW.getId()
```

Comment:
As this equals comparison is quite lengthy, potentially expensive through its string replace operation and occurs twice, I wonder if you could simply do it once in initialize() and set a flag within the handler. In the handleCommand, you would then only have to check the flag.